### PR TITLE
Gradle 6.9, Android Gradle Plugin 4.2.1

### PR DESCRIPTION
**Pull request by dulmandakh**
## Summary

Bump Gradle to 6.9 which supports Apple Silicon, also Android Gradle Plugin 4.2.1 which defaults to Java 1.8 so no additional config required.

## Changelog

[Android] [Changed] - Bump Gradle to 6.9, Android Gradle Plugin to 4.2.1

## Test Plan

rn-tester builds and runs as expected

### CODE REVIEW

**Comment by dulmandakh** [comment]
@fkgozali @mdvacca please review and merge

**Comment by dulmandakh** [comment]
it's almost green 😅

**Comment by ShikaSD** [comment]
@facebook-github-bot import

File: packages/react-native-gradle-plugin/build.gradle.kts [hunk 1]
```diff
@@ -8,7 +8,7 @@
 plugins {
   `java-gradle-plugin`
   `kotlin-dsl`
-  kotlin("jvm") version "1.4.21"
```
Comment (liamjones) [inline packages/react-native-gradle-plugin/build.gradle.kts:11]: This was a deliberate downgrade?

I don't have any specific knowledge, just noticed it was the only number going down not up! 😅

Comment (dulmandakh) [reply packages/react-native-gradle-plugin/build.gradle.kts:11]: it was printing warning, so removed it 😅

**Comment by dulmandakh** [comment]
@mdvaccaplease help with a failing test. Anything I can help?

**Comment by ShikaSD** [comment]
The test seems to be unrelated, I will update it separately
There are some internal things we need to update before merging, I'll try to look into them soon-ish :)

**Pull request closed without merging**